pkg/machines/components/vmCloneDialog.jsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import { Button, Form, FormGroup, Modal, TextInput } from '@patternfly/react-core';
+
+import { isEmpty, isObjectEmpty } from '../helpers.js';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+import "./vmCloneDialog.scss";
+const _ = cockpit.gettext;
+
+export const CloneDialog = ({ name, connectionName, toggleModal }) => {
+    const [newVmName, setNewVmName] = useState(name + '-clone');
+    const [validated, setValidated] = useState(false);
+    const [inProgress, setInProgress] = useState(false);
+    const [virtCloneOutput, setVirtCloneOutput] = useState('');
+    const [error, dialogErrorSet] = useState({});
+
+    function validateParams() {
+        const validation = {};
+        if (isEmpty(newVmName.trim()))
+            validation.name = _("Name must not be empty");
+
+        return validation;
+    }
+
+    function onClone() {
+        setValidated(true);
+
+        const validation = validateParams();
+        if (!isObjectEmpty(validation)) {
+            setInProgress(false);
+            return;
+        }
+
+        setInProgress(true);
+        return cockpit.spawn(["virt-clone", "--connect", "qemu:///" + connectionName, "--original", name, "--name", newVmName, "--auto-clone"], { superuser: "try" })
+                .stream(setVirtCloneOutput)
+                .then(toggleModal, exc => {
+                    setInProgress(false);
+                    dialogErrorSet({ dialogError: cockpit.format(_("Failed to clone VM $0"), name), dialogErrorDetail: exc.message });
+                });
+    }
+
+    const validationFailed = validateParams();
+    return (
+        <Modal position="top" variant="medium" isOpen onClose={toggleModal}
+           title={cockpit.format(_("Create a clone VM based on $0"), name)}
+           footer={
+               <>
+                   {!isObjectEmpty(error) && <ModalError dialogError={error.dialogError} dialogErrorDetail={error.dialogErrorDetail} />}
+                   {virtCloneOutput && <code className="vm-clone-virt-clone-output">{virtCloneOutput}</code>}
```

Comment:
The error handling isn't really useful here. When I try to cause an error by inputting the name of an existing VM, I get an error: 
```
Failed to clone VM fedora32
Error message: virt-clone exited with code 1 
```
Which isn't really useful.

When I run the same in CLI, I get an output:
`ERROR    Invalid name for new guest: Guest name 'rww' is already in use.`
Which is much more useful and that's what we want to show in a modal.

I guess that's what the `virtCloneOutput` is there for. That right now returns and empty string. Adding a `pty` parameter to cockpit.spawn should fix that ;).

Reply:
Fixed.